### PR TITLE
Implement FSDPA-based prompt attention

### PR DESCRIPTION
**Pull request by mdvoretc-intel**

### CODE REVIEW

**Comment by mdvoretc-intel** [comment]
@kzawora-intel Confirmed functional on batch 1. Batches of mixed-length sequences require additional testing to verify suitability of triangular masking.

**Comment by mdvoretc-intel** [comment]
The algorithm for FSDPA doesn't expect the sequence stacking hinted at in the current prompt attention API, making it incompatible with existing unit tests.

File: vllm/model_executor/layers/attention.py [hunk 1]
```diff
@@ -165,9 +172,15 @@ def forward(
                     p=0.0,
                     scale=self.scale,
                 )
-                output = torch.zeros_like(query)
-                output[:, :out.shape[1], :, :] = out
-                output = output.view_as(query)
+                if FusedSDPA:
+                    if out.shape[0] * out.shape[1] * out.shape[2] * out.shape[3] != batch_size * seq_len * hidden_size:
+                        import pdb
+                        pdb.set_trace()
```
Comment (tkrupa-intel) [inline vllm/model_executor/layers/attention.py:178]:
```suggestion
```

File: vllm/hpu/xops.py [hunk 2]
```diff
@@ -39,29 +45,45 @@ def memory_efficient_attention_forward(
...
-        mask_end_idx = (i + 1) * seq_len
+    if FusedSDPA:
+        _, _, heads, head_dim = query.shape
+        bs = len(cu_seq_lens)
+        seq_len_q = query.shape[1] // bs
+        seq_len_kv = key.shape[1] // bs
+        query = query.reshape(bs, seq_len_q, heads, head_dim).permute(0, 2, 1, 3)
+        key = key.reshape(bs, seq_len_kv, heads, head_dim).permute(0, 2, 1, 3)
+        value = value.reshape(bs, seq_len_kv, heads, head_dim).permute(0, 2, 1, 3)
+        attn_mask = torch.ones((bs, heads, seq_len_q, seq_len_kv), device=query.device, dtype=torch.bool).tril()
```
[the remark this answers is not in the document]
Comment (kzawora-intel) [reply vllm/hpu/xops.py:56]: probably would be a good idea to fall back on the stock implementation if attn_bias is not None

[the remark this answers is not in the document]
Comment (mdvoretc-intel) [reply vllm/hpu/xops.py:57]: Updated.
